### DISCUSSION TITLE
Is there an Setup outline that explains the steps needed to set this up, without diving into the detail?

### DISCUSSION THREAD

**Discussion by douginoz** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#617] [Q&A] [answered]
I'm having a go at this but the Configuration section only says, "This card contains UI configuration editor, but it is limited to only basic set of features." .  There's a link to an example demo config.  
But then the rest of the instructions don't actually say what to do - they are just tables of configuration keys, values, and descriptions. 

Here's the problem - it's written by people with deep, deep understanding of the underlying details of the system, but that's no help to someone wanting to use this.  I need to find answers to things like:

- I've installed the card and added it to my dashboard.  I've filled in the available fields - "Title", "Vacuum entity", and "Vacuum platform".  Now what? (What are the next steps I must to?)
- Since I'm using a Roomba, I have created a map .png file of my house and I've added it's path to the map source: image:  and I can now see it in the card.  So what are the next steps with this?  It does nothing.  I can't do anything with this card.  I can't manually select regions of the image and assign area or room names to them. 

The example demo config is full of interesting things, but there's no instructions on what needs to be added, when, and how.  
It could take me several days of reading all the configuration keys, values, etc and try to figure out which ones I need to use, and why.  

There really needs to be a high-level installation guide, something that first explains at a high level what needs to be done, then references all that data and configuration keys etc. if needed.  For example:

1. Once you create the card in your dashboard, you need to fill in the fields x, y, z
2. Then you will need to add an image to the HA folder and include it's path etc
3. Then you will need to map out the image into zones or areas.  This is done by doing  (a) (b) then step (c) etc.
4. Once you have your zones created, you'll want to add icons or hotspots or something so that they can be selected when using the card.  Any spot that's selected will tell the vacuum where to clean.  To add hotspots, do (a), (b), ... 

For me anyway, it's just too overwhelming to try to read over 100 configuration keys, values, etc. and try to work out which ones I'm supposed to be using, and how to use them.  It's kind of like being given a model plane kit, and the only instructions that came with it are a list of all the parts and what they do - "the rotor actuator delivers forward momentum when given positive torque" - but no actual instructions on how to build the plane!

5

**Comment by PiotrMachowski** [comment]
Hi, @douginoz!
Thank you for your input. The problem with this card is that is very highly configurable, which might make things quite overwhelming.

Have you found a [documentation for Roomba vacuum platform](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/blob/master/docs/templates/roomba.md)? It should be linked in the card editor as well.

The part matters the most is `map_modes`, the rest should be configured completely automatically.

One more thing: when changing to a setup platform (to get coordinates) you have to use following config:
```yaml
calibration_source:
  platform: Roomba
```

**Reply by douginoz** [reply, marked as answer]
Hi Peter,
Thanks for the reply. 

 I've made  progress.  I manually edited the YAML to include the following:
calibration_source:
  identity: true

And now I can see the Zone cleanup drop-down icon below the map.  

This might be the big stumbling block - there's nothing I read that told me I have to manually edit the YAML to include this.  I think that needs to be made clear that this is a step that needs to be done before I can use the tool to get screen image coordinates.

The other missing step is that I must set `vacuum_platform: Setup integer` FIRST so that I can use the drawing box function that returns the room coordinates that I can later use in the YAML.  Perhaps I missed this somewhere but if not, I think it needs to be made clear.   Then once the coordinates are all worked out, the value needs to be set back to `vacuum_platform: Roomba`, otherwise it returns an 'unknown template' error when you try to enter 

```
map_modes:
  - template: vacuum_clean_segment
```

I'm now editing the YAML to include all the rooms.  Looks like progress!
Many thanks

**Reply by PiotrMachowski** [reply]
> I've made progress. I manually edited the YAML to include the following:
calibration_source:
identity: true

This part shouldn't be necessary if you use `vacuum_platform: Roomba`, but when you change to `vacuum_platform: Setup integer` you should use following code (that I have mentioned in the previous comment):
```yaml
calibration_source:
  platform: Roomba
```
In case of Roombas it does exactly the same as `identity: true`, but it might be different for other vacuums.


Generally speaking a documentation would benefit from some major updates, especially for users that use non-Xiaomi vacuums (card is configured completely automatically for them).
